pilot/pkg/model/jwks_resolver.go

```diff
@@ -435,3 +435,41 @@ func (r *JwksResolver) refresh() {
 func (r *JwksResolver) Close() {
 	closeChan <- true
 }
+
+func compareJWKSResponse(oldKeyString string, newKeyString string) bool {
+	var oldJWKs map[string]interface{}
+	var newJWKs map[string]interface{}
+	e := json.Unmarshal([]byte(oldKeyString), &oldJWKs)
+	if e != nil {
+		log.Warnf("Previous JWKs public key JSON is not parseable: %s", oldKeyString)
+		return true
+	}
+	e = json.Unmarshal([]byte(newKeyString), &newJWKs)
+	if e != nil {
+		log.Warnf("New JWKs public key JSON is not parseable: %s", newKeyString)
+		return true
```

Comment:
Should we return "false" here, or better, error here? We shouldn't update the key if there is parsing error, right?

Reply:
Will modify to better handle this and return an error if the new key is not parseable